src/main/java/com/conveyal/r5/profile/FastRaptorWorker.java

```diff
@@ -0,0 +1,559 @@
+package com.conveyal.r5.profile;
+
+import com.conveyal.r5.transit.TransitLayer;
+import com.conveyal.r5.transit.TripPattern;
+import com.conveyal.r5.transit.TripSchedule;
+import gnu.trove.list.TIntList;
+import gnu.trove.list.array.TIntArrayList;
+import gnu.trove.map.TIntIntMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * RaptorWorker is fast, but FastRaptorWorker is knock-your-socks-off fast, and also more maintainable.
+ * It is also simpler; all propagation code has been removed and will be performed elsewhere if desired.
+ *
+ * The algorithm used herein is described in
+ *
+ * Conway, Matthew Wigginton, Andrew Byrd, and Marco van der Linden. “Evidence-Based Transit and Land Use Sketch Planning
+ *   Using Interactive Accessibility Methods on Combined Schedule and Headway-Based Networks.” Transportation Research
+ *   Record 2653 (2017). doi:10.3141/2653-06.
+ *
+ * Delling, Daniel, Thomas Pajor, and Renato Werneck. “Round-Based Public Transit Routing,” January 1, 2012.
+ *   http://research.microsoft.com/pubs/156567/raptor_alenex.pdf.
+ *
+ * There is currently no support for saving paths.
+ */
+public class FastRaptorWorker {
+    private static final Logger LOG = LoggerFactory.getLogger(FastRaptorWorker.class);
+
+    /** Step for departure times */
+    private static final int DEPARTURE_STEP_SEC = 60;
+
+    /** Minimum wait for boarding to account for schedule variation */
+    private static final int MINIMUM_BOARD_WAIT_SEC = 60;
+
+    // Variables to track time spent, all in nanoseconds (some of the operations we're timing are significantly submillisecond)
+    // (although I suppose using ms would be fine because the number of times we cross a millisecond boundary would be proportional
+    //  to the portion of a millisecond that operation took).
+    public long startClockTime;
+    public long timeInScheduledSearch;
+    public long timeInScheduledSearchTransit;
+    public long timeInScheduledSearchFrequencyBounds;
+    public long timeInScheduledSearchTransfers;
+
+    public long timeInFrequencySearch;
+    public long timeInFrequencySearchFrequency;
+    public long timeInFrequencySearchScheduled;
+    public long timeInFrequencySearchTransfers;
+
+    /** the transit layer to route on */
+    private final TransitLayer transit;
+
+    /** Times to access each transit stop using the street network (seconds) */
+    private final TIntIntMap accessStops;
+
+    /** The profilerequest describing routing parameters */
+    private final ProfileRequest request;
+
+    /** Frequency-based trip patterns running on a given day */
+    private TripPattern[] runningFrequencyPatterns;
```

Comment:
No, it doesn't block the merge. Just observations as I was reviewing. I'll create a ticket for this after the merge.